script.rb

```diff
@@ -0,0 +1,148 @@
+module Enumerable
+  def my_each
+    return to_enum(:my_each) unless block_given?
+
+    i = 0
+    until i == size
+      yield self[i]
+
+      i += 1
+    end
+    self
+  end
+
+  # my_each_with_index
+
+  def my_each_with_index
+    return to_enum(:my_each) unless block_given?
+
+    i = 0
+    until i == size
+      yield(self[i], i)
+      i += 1
+    end
+    self
+  end
```

Comment:
- [ ] `my_each_with_index` is not working as the original `each_with_index` method, also is not passing all test cases.

### `each_with_index` is an Enumerable that has these characteristics: 

- it returns the array/range/hash itself if a block is given.
```
[1, 2, 3].each_with_index { |num, idx| num }
# [1, 2, 3]
(1..3).each { |num, idx| num * idx }
# 1..3
````
- it applies the block for each element inside the array/range/hash
```
[1, 2, 3].each_with_index { |num, idx| puts num, idx }
# 1
# 0
# 2
# 1
# 3
# 2
# => [1, 2, 3]
```
[Enumerator#each](https://ruby-doc.org/core-2.7.2/Enumerable.html#method-i-each_with_index)